### PR TITLE
Applies a range check in converting severity and facility code indexes before accessing the lookup array

### PR DESCRIPTION
**Pull request by andsel**
## Release notes

Fixes range check problem adding validation of severity and facility codes extracted from `syslog_pri_field`.

## What does this PR do?
Adds a range check when accessing the lookups for code to label translation of severity and facility. When the field `syslog_pri_field` contains a numerical value that's bigger then the Long type's max value, a RangeError happened.
This commit verifies if those code indices are bigger than such max value and default the string translation to empty.

## Why is it important/What is the impact to the user?
As described in #12 when a too much big value is inserted into `syslog_pri_field` then the filter crashes the pipeline with `RangeError`. However, when a code couldn't be looked up into a string an empty string is assigned as label description of the code. This change applies the same logic, a index that doesn't have a matching label is empty-labelled.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files (and/or docker env variables)~~
- [x] I have added tests that prove my fix is effective or that my feature works

## Author's Checklist

- [x] run with #12 issue example pipeline.

## How to test this PR locally

Execute the steps described in #12

## Related issues

- Related #12

### CODE REVIEW

File: lib/logstash/filters/syslog_pri.rb [hunk 1]
```diff
@@ -105,11 +105,20 @@ def parse_pri(event)
...
+      severity_label = retrieve_with_range_check(@severity_labels, severity_code, 'severity_code')
       event.set(@severity_label_key, severity_label) if severity_label
     end
   end # def parse_pri
+
+  def retrieve_with_range_check(array, index, property_name)
+    if index > Java::JavaLang::Long::MAX_VALUE
+      logger.warn("Extracted #{property_name} too big", property_name.to_sym => index, :max_value => Java::JavaLang::Long::MAX_VALUE)
+      nil
+    else
```
Comment (mashhurs) [inline lib/logstash/filters/syslog_pri.rb:121]: `facility_labels` and `severity_labels` arrays have limited (13 & 8) elements. I would prefer to precisely tell if requested index isn't in array size range. Does it make sense?
```suggestion
    if index >= array.size
      level_name = property_name == 'facility_code'? 'Facility' : 'Severity'
      logger.warn("#{level_name} labels do not contain value for #{index} index.")
      nil
    else
```

Comment (andsel) [reply lib/logstash/filters/syslog_pri.rb:121]: Today that log is not printed, when an index doesn't fall into the array it's not translated to label, without any logging.
Moving to the `log.warn` for each index that fall outside the array could generate a lot of noisy logs I think.

**Review by mashhurs — APPROVED** [review]
LGTM, solves `RangeError`!

File: lib/logstash/filters/syslog_pri.rb [hunk 2]
```diff
@@ -105,11 +105,21 @@ def parse_pri(event)
 
     # Add human-readable names after parsing severity and facility from PRI
     if @use_labels
```
Comment (jsvd) [inline lib/logstash/filters/syslog_pri.rb:107]:
```ruby
    return unless @use_labels
    
    if facility_label = @facility_labels[facility_code]
      event.set(@facility_label_key, facility_label)
    else
      event.tag('_syslogparsefailure')
      logger.debug? && logger.debug("Invalid facility code for event", :facility => facility_code)
      return
    end
    
    # severity code is inrange [0..7] by definition, no need to check any bound
    severity_label = @severity_labels[severity_code]
    event.set(@severity_label_key, severity_label) if severity_label
```

Comment (andsel) [reply lib/logstash/filters/syslog_pri.rb:107]: I'll integrate this in #11 and close this as superseeded by it.

**Comment by andsel** [comment]
Close as superseeded by #11

**Pull request closed without merging**